TypeScript error in llms-full.txt/route.ts (array map callback type mismatch)

## TypeScript Error in llms-full.txt/route.ts

### Problem
TypeScript type error exists in `apps/web/app/llms-full.txt/route.ts` that is currently masked by `ignoreBuildErrors` setting.

### Affected File & Error

**File:** `apps/web/app/llms-full.txt/route.ts`

**Line 69:**
```
error TS2345: Argument of type '(c: { name: string; }) => string' is not assignable to parameter of type '(value: Record<string, any>, index: number, array: Record<string, any>[]) => string'.
  Types of parameters 'c' and 'value' are incompatible.
    Type 'Record<string, any>' is not assignable to type '{ name: string; }'.
```

### Impact
- Type safety is compromised in the llms-full.txt API route
- Potential runtime errors if the collection objects don't have the expected `name` property
- Indicates a mismatch between the actual data structure and the expected type

### Suggested Fix

The issue is that the map callback expects `{ name: string }` but the array contains `Record<string, any>`.

**Option 1: Add type guard/assertion**
```tsx
// Before:
collections.map((c) => c.name)

// After:
collections.map((c: { name: string }) => c.name)
// Or:
collections.map((c) => (c as { name: string }).name)
```

**Option 2: Fix the source type**
If `collections` should be typed as `Array<{ name: string }>`, update the query or type definition:
```tsx
const collections: Array<{ name: string }> = await fetchCollections();
```

**Option 3: Add runtime validation**
```tsx
collections
  .filter((c): c is { name: string } => typeof c.name === 'string')
  .map((c) => c.name)
```

### Related
- #2294 (Critical: ignoreBuildErrors masks TypeScript errors in production builds)
- #2306 (SEO: Enhance llms.txt with collection URLs and structured data for AI search optimization)

--- a/apps/docs/app/llms-full.txt/route.ts
+++ b/apps/docs/app/llms-full.txt/route.ts
@@ -1,6 +1,6 @@
 import { buildApiDocLLMText, buildApiOverviewLLMText } from '@/lib/api-llm-text';
 import { getLLMText } from '@/lib/get-llm-text';
-import { getApiDoc, getApiDocs, getApiOverview } from '@/lib/content';
+import { type ApiDoc, getApiDoc, getApiDocs, getApiOverview } from '@/lib/content';
 import { source } from '@/lib/source';
 
 export const revalidate = false;
@@ -15,7 +15,7 @@ export async function GET() {
   const pages = [
     ...docsPages,
     ...(apiOverview ? [buildApiOverviewLLMText(apiOverview)] : []),
-    ...apiPages.filter((page) => page != null).map(buildApiDocLLMText),
+    ...apiPages.filter((page): page is ApiDoc => page != null).map(buildApiDocLLMText),
   ];
 
   return new Response(pages.join('\n\n'), {

